File: DSA/DSA03012 SẮP ĐẶT XÂU KÝ TỰ 1.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    string s;
    cin >> s;

    vector<int> alpha(26);
    int maxx = 0;
    for (auto &&i : s)
    {
        alpha[i-'a']++;
        maxx = max(maxx, alpha[i-'a']);
    }
    int max = (s.length()+1)/2;
    
    maxx <= max ? (cout << 1) : (cout << -1);
    
    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}

File: DSA_python/DSA03012 SẮP ĐẶT XÂU KÝ TỰ 1.py
def submit():
    s = input()
    alpha = [0]*26
    maxx = 0

    for i in range(len(s)):
        alpha[ord(s[i])-ord('a')] += 1
        maxx = max(maxx, alpha[ord(s[i])-ord('a')])
    maxi = (len(s)+1)/2

    print(1 if maxx <= maxi else -1)

test = int(input())
for i in range(test):
    submit()